packages/esm-service-queues-app/src/send-patient-back-toqueue/send-back-patient-toqueue.component.tsx

```diff
@@ -0,0 +1,288 @@
+  InlineLoading,
+  Tile,
+  TextInput,
+} from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import { navigate, showSnackbar, useConfig } from '@openmrs/esm-framework';
+import { useQueueLocations } from '../patient-search/hooks/useQueueLocations';
+import styles from './send-back-patient-toqueue.scss';
+import { useQueues } from '../hooks/useQueues';
+import { useForm, Controller } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { type ConfigObject } from '../config-schema';
+import { useMutateQueueEntries } from '../hooks/useQueueEntries';
+import { useQueueEntry } from './send-back-patient-toqueue.resource';
+import { updateQueueEntry } from '../active-visits/active-visits-table.resource';
+
+interface SendBackPatientProps {
+  patientUuid: string;
+  closeModal: () => void;
+}
+
+const SendBackPatientToQueue: React.FC<SendBackPatientProps> = ({ closeModal, patientUuid }) => {
+  const { t } = useTranslation();
+  const { concepts } = useConfig<ConfigObject>();
+  const { data, error, isLoading } = useQueueEntry(patientUuid);
+  const allowedPriorities = data?.queue?.allowedPriorities ?? [];
+  const allowedStatuses = data?.queue?.allowedStatuses ?? [];
+  const uppercaseText = (text: string) => text.toUpperCase();
+  const defaultWaitStatus = concepts?.defaultStatusConceptUuid;
+  const defaultResultReviewUrgencyStatus = concepts?.defaultResultReviewPriorityConceptUuid;
+
+  const schema = useMemo(
+    () =>
+      z.object({
+        location: z.string({ required_error: t('queueLocationRequired', 'Queue location is required') }),
+        service: z.string({ required_error: t('serviceIsRequired', 'Service is required') }),
+        status: z.string({ required_error: t('statusIsRequired', 'Status is required') }),
+        priority: z.string({ required_error: t('priorityIsRequired', 'Priority is required') }),
+      }),
+    [t],
+  );
+
+  type ChangeStatusForm = z.infer<typeof schema>;
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+    getValues,
+  } = useForm<ChangeStatusForm>({
+    resolver: zodResolver(schema),
+  });
+
+  const queueServiceLocation = data?.locationWaitingFor?.display ?? '';
+  const { queues } = useQueues(queueServiceLocation);
+  const { queueLocations } = useQueueLocations();
+  const { mutateQueueEntries } = useMutateQueueEntries();
+
+  const onSubmit = async (updateQueueData: ChangeStatusForm) => {
+    const { status, service } = updateQueueData;
+    const defaultPriority = concepts.defaultPriorityConceptUuid;
+    const queuePriority = updateQueueData.priority === '' ? defaultPriority : updateQueueData.priority;
+    const emergencyPriorityConceptUuid = concepts.emergencyPriorityConceptUuid;
+    const sortWeight = updateQueueData.priority === emergencyPriorityConceptUuid ? 1.0 : 0.0;
+    const endDate = new Date();
+
+    try {
+      const response = await updateQueueEntry(
```

Comment:
This is an older `updateQueueEntry` function that should probably be deprecated and not used.

What do we expect the user's queue entry status to be when we open this modal? Do they already have a queue entry? If so, we should end their current queue entry and create a new one. This action is referred to as a  "Transiion". (Confusingly, there is an older `transition-entry.component.tsx` file that does something else)

If we expect the patient to not have an active queue entry, we should be creating a new queue entry.

Reply:
Thanks for this clarification

Reply:
@ojwanganto @chibongho I want to update the video for the PR we see both scenarios have be catered for if there is a suggestion from there I can pick it up